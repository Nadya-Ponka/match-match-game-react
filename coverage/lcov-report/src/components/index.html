<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.96% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>71/96</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.32% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>44/95</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>20/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/15</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Button.js"><a href="Button.js.html">Button.js</a></td>
	<td data-value="70.59" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 70%;"></div><div class="cover-empty" style="width:30%;"></div></div></td>
	<td data-value="70.59" class="pct medium">70.59%</td>
	<td data-value="34" class="abs medium">24/34</td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	</tr>

<tr>
	<td class="file high" data-value="Something.js"><a href="Something.js.html">Something.js</a></td>
	<td data-value="87.88" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.88" class="pct high">87.88%</td>
	<td data-value="33" class="abs high">29/33</td>
	<td data-value="57.58" class="pct medium">57.58%</td>
	<td data-value="33" class="abs medium">19/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="Timer.js"><a href="Timer.js.html">Timer.js</a></td>
	<td data-value="62.07" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 62%;"></div><div class="cover-empty" style="width:38%;"></div></div></td>
	<td data-value="62.07" class="pct medium">62.07%</td>
	<td data-value="29" class="abs medium">18/29</td>
	<td data-value="37.93" class="pct low">37.93%</td>
	<td data-value="29" class="abs low">11/29</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 28 2018 23:09:26 GMT+0200 (Романское время (лето))
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
<script src="..\..\block-navigation.js"></script>
</body>
</html>
