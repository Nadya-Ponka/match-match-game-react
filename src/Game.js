import React, { Fragment } from 'react';
import { Route } from 'react-router';

import '../Styles/style.css';

import LandingPage from './components/landing-page/landing-page';
import Introduction from './components/introdaction/Introdaction';
import PlayingField from './components/playingField/playingField';

require('babel-core/register');
require('babel-polyfill');

class GameDOM extends React.Component {
    render() {
        return (
            <Fragment>
                <section className="greeting">
                    <LandingPage />
                    <Introduction/>
                </section>
                <main>
                    <Route path='/' component={PlayingField} />
                </main>
            </Fragment>
        );
    }
}

export default GameDOM;
