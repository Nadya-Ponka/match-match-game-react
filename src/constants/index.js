export const CHOOSE_BACK_CARD = 'CHOOSE_BACK_CARD';
export const CHOOSE_DIFFICULTY = 'CHOOSE_DIFFICULTY';
export const MAKE_CARD_ARRAY = 'MAKE_CARD_ARRAY';
export const START_TIMER = 'START_TIMER';
export const STOP_TIMER = 'STOP_TIMER';
export const FLIP_CARD = 'FLIP_CARD';
export const MATCH_FOUND = 'MATCH_FOUND';
export const SET_LAST_FLIPPED_CARD = 'SET_LAST_FLIPPED_CARD';
export const SHOW_LEADERBOARD = 'SHOW_LEADERBOARD';
