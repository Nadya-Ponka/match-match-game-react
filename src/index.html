<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>

    <link rel="stylesheet" href="../Styles/style.css" type="text/css">

</head>
<body>
    <div id="root">
    </div>
    <script async type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js"></script>
	<script async src="https://code.jquery.com/jquery-1.12.4.js"></script>
	<script async src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
</body>
</html>