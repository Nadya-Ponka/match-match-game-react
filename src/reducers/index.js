import { combineReducers } from 'redux';

import leaderBoard from './leaderBoardReducer';
import { cards, backCard, level } from './prepareDataReducer';
import { startedAt, stoppedAt } from './timerReducer';
import { flipped, matches, lastCard } from './gameReducer';

export default combineReducers({
    backCard,
    level,
    startedAt,
    stoppedAt,
    flipped,
    matches,
    lastCard,
    leaderBoard,
    cards,
});
