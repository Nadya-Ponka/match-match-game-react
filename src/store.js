import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from './reducers';
import ping from './middleWare/ping';

const store = createStore(reducer, applyMiddleware(thunk, ping));

export default store;
