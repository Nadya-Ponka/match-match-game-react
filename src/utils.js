require('babel-core/register');
require('babel-polyfill');
/* export async function getScoreFromAPI () {
        const response = await fetch('http://mmg-score.herokuapp.com');
        return await response.json().then(function (myJson){return myJson});
}*/
export let period;

// leaderBoard
export let displayed;

function sortByField(arr, field) {
    function byField(a, b) {
        if (a[field] >= b[field]) return 1;
        if (a[field] < b[field]) return -1;
    }
    return arr.sort(byField);
}

export function secondsInTimeFormat(s) {
    function num(val) {
        val = Math.floor(val);
        return val < 10 ? `0${val}` : val;
    }
    const hours = (s / 3600) % 24;
    const minutes = (s / 60) % 60;
    const seconds = s % 60;
    return `${num(hours)}:${num(minutes)}:${num(seconds)}`;
}

export function leaderBoard(totalScore) {
    const userData = {
        username: document.querySelector('#firstName').value,
        email: document.querySelector('#email').value,
        score: totalScore,
        // tab: gridSize
    };
    const serialObj = JSON.stringify(userData);
    console.log('userData.score'+userData.score);

    (async () => {
        const rawResponse = await fetch('http://mmg-score.herokuapp.com', {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
            },
            body: serialObj,
        });

        const dataBase = [];
        let dataFromAPI;

        fetch('http://mmg-score.herokuapp.com')
            .then(response => response.json())
            .then((myJson) => {
                dataFromAPI = myJson;
                const data = dataFromAPI.result;
                const keysLocalStorage = Object.keys(data);
                keysLocalStorage.forEach((key) => {
                    const returnedObj = data[key];
                    dataBase.push(returnedObj);
                });

                //setTimeout(() => {
                    const sorted = sortByField(dataBase, 'score');
                    displayed = sorted.splice(0, 55);
                    console.log(`displayed:   ${displayed}`);
                //}, 3000);
            })
            .catch(new Error('Something gone wrong'));
    })();
}

// Fisher--Yates Algorithm перемешивания массива рандомом
export function shuffle(arrayForShuffle) {
    let counter = arrayForShuffle.length;
    let temp;
    let index;
    // While there are elements in the array
    while (counter > 0) {
    // Pick a random index
        index = Math.floor(Math.random() * counter);
        counter--;
        // And swap the last element with it
        temp = arrayForShuffle[counter];
        arrayForShuffle[counter] = arrayForShuffle[index];
        arrayForShuffle[index] = temp;
    }
    return arrayForShuffle;
}

export function sendDataToAPI(totalScore) {
    const userData = {
        username: document.querySelector('#firstName').value,
        email: document.querySelector('#email').value,
        score: totalScore,
        // tab: gridSize
    };
    const serialObj = JSON.stringify(userData);
    console.log('userData.score'+userData.score);

    (async () => {
        const rawResponse = await fetch('http://mmg-score.herokuapp.com', {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
            },
            body: serialObj,
        });
    })();
}
