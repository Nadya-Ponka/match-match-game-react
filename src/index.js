import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router';
import { BrowserRouter } from 'react-router-dom';

import store from './store';
import GameDOM from './Game';

ReactDOM.render(
    <Provider store={store}>
        <BrowserRouter history={browserHistory}>
            <GameDOM />
        </BrowserRouter>
    </Provider>,
    document.querySelector('#root'),
);
