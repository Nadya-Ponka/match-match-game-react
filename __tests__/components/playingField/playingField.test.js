import React from 'react';
import { shallow } from 'enzyme';

import ConnectedPlayingField, { PlayingField } from '../../../src/components/playingField/playingField';
import { leaderBoard } from '../../../src/utils';

describe('PlayingField', () => {
    const state = {
        cards: [],
        level: '5',
        leaderBoard: [],
        subscribe: () => {},
        dispatch: () => {},
        getState: () => {},
    };

    test('match snapshot', () => {
        const wrapper = shallow(<PlayingField store={state} />);
        expect(wrapper).toMatchSnapshot();
    });
});
