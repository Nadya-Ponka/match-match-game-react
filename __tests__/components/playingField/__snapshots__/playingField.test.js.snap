// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayingField match snapshot 1`] = `
<React.Fragment>
  <section
    className="playingField"
  >
    <Connect(CardsField) />
    <div
      className="win"
    >
      <p
        className="winner"
      >
        Ура! Все пары найдены!
        <br />
        Твое время:
        <span
          className="timer"
          id="score"
        />
      </p>
      <div
        className="router"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="buttonRouter"
          id="resultsTable"
          to="leader-board"
        >
          Топ-10
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="buttonRouter"
          to="smth-else"
        >
          Info
        </NavLink>
        <Route
          component={[Function]}
          path="/leader-board"
        />
        <Route
          component={[Function]}
          path="/smth-else"
        />
      </div>
      <Button
        className="buttonStart"
        onClick={[Function]}
        text="Играть еще"
      />
    </div>
  </section>
</React.Fragment>
`;
